client_it_test.go

```diff
@@ -213,3 +214,95 @@ func TestClient_AddDistributedObjectListener(t *testing.T) {
 		mu.Unlock()
 	})
 }
+
+func TestClusterReconnection(t *testing.T) {
```

Comment:
Maybe you can add a map.put test while cluster is restarted. What happens in that case? Is map.put synchronous? will it work after cluster restart?

Reply:
map.put is synchronous. But it's a bit hard to test it with the current setup since we shut down the cluster in order to be able to trigger client disconnection. So, if that happens right before map.put, a `nil` value is returned instead of the old value since that information is lost when the cluster is down. We need a way to cause an arbitrary disconnection between the client and cluster. I'll think about that after V1.